src/synth/fluid_mod.c

```diff
@@ -334,21 +334,21 @@ fluid_mod_transform_source_value(fluid_real_t val, unsigned char mod_flags, cons
      * is close enough.
      */
     case FLUID_MOD_SIN | FLUID_MOD_UNIPOLAR | FLUID_MOD_POSITIVE: /* custom sin(x) */
-        val = sin(M_PI / 2 * val_norm * 0.87);
+        val = sin(M_PI_2 * val_norm * 0.87);
```

Comment:
Aha, you want to avoid promotion of the literals to double, causing the whole expression to be evaluated as double, although `WITH_FLOAT` is defined. But then you could have done:

```suggestion
        val = sin((M_PI / 2.0f) * val_norm * 0.87f);
```

Reply:
This change did not removed the double calculation from my code because M_PI. Unfortunately, I had to change it to something like this:

`val = sin((fluid_real_t)(M_PI / 2.0  * 0.87) * val_norm);`

When compiling for an ARM Cortex with VFPv4-SP FPU, which supports only single precision numbers, this effect was made very visible.

Reply:
Sure, if M_PI is double, it would be promoted to double again. Perhaps this explicit cast as you said
 
```
val = sin((fluid_real_t)(FLUID_M_PI / 2.0 * 0.87) * val_norm);
```

would be somewhat strange / redundant / hard to read if `FLUID_M_PI` is defined as cast itself:

```
val = sin((fluid_real_t)(((fluid_real_t)M_PI) / 2.0 * 0.87) * val_norm);
```

So, I have a slight preference for:

```
val = sin((FLUID_M_PI / 2.0f * 0.87f) * val_norm);
```

Btw, well spotted to place the 0.87 next to PI to allow proper constant folding and saving one multiplication at runtime.